### PR TITLE
`text_blobs`/Text module support for service worker format in local mode

### PR DESCRIPTION
**Pull request by threepointone**
This adds support for `text_blobs`/Text module support in local mode. Now that https://github.com/cloudflare/miniflare/pull/228 has landed in miniflare (thanks @caass!), we can use that in wrangler as well.

Fixes https://github.com/cloudflare/wrangler2/issues/416

--- 

We can land this before miniflare does a release, but it won't work until we update miniflare. // cc @mrbbot

### CODE REVIEW

**Comment by threepointone** [comment]
Don't land this just yet, I'm doing some manual testing here. Please review tho!

File: packages/wrangler/src/create-worker-upload-form.ts
Comment (petebacondarwin) [inline packages/wrangler/src/create-worker-upload-form.ts:178]: Is this change specific to this PR? It kind of feels like it is an independent fix? But I might be misunderstanding its importance in terms of text_blobs?

**Review by petebacondarwin — APPROVED** [review]
Looking good. We'll wait to see the result of the manual testing before merging.

File: packages/wrangler/src/create-worker-upload-form.ts
Comment (threepointone) [reply packages/wrangler/src/create-worker-upload-form.ts:178]: This just rounds out the story for module system support. With module workers, we can import modules of all the supported types: js, wasm, text, and data. With service-workers, we can import js, wasm, text, but _not_ data. We can detect its usage and throw an error. I simply moved the error to earlier in the pipeline, right when we're building the worker, instead of waiting for when we create the worker form upload. 

I don't expect this error to trigger often; there are no service-worker workers right now that import a Data module (because there's no equivalent like `wasm_modules` or `text_blobs` for data.) The usecase this will probably trigger is when a third party library imports a Data module, and is used in a service-worker (It'll work fine with a module worker). If that happens, and it's a good/common library, then the thing to do will be to convince the runtime team to add a new type of binding (`data_buffers`?). We can then leverage that to add support for Data modules for service-workers.

Comment (threepointone) [reply packages/wrangler/src/create-worker-upload-form.ts:178]: I just checked internally, and we _do_ have support for a `data_blob` binding. Filed https://github.com/cloudflare/wrangler2/issues/740

**Pull request merged**
